packages/core/src/Actor.ts

```diff
@@ -43,14 +44,21 @@ export function createNullActor(id: string): Actor {
  */
 export function createInvocableActor<TC, TE extends EventObject>(
   invokeDefinition: InvokeDefinition<TC, TE>,
-  machine: StateMachine<TC, any, TE>
+  machine: StateMachine<TC, any, TE>,
+  context: TC,
+  _event: SCXML.Event<TE>
 ): Actor {
   const tempActor = createNullActor(invokeDefinition.id);
   const serviceCreator = machine.options.services?.[invokeDefinition.src];
   tempActor.deferred = true;
 
   if (isMachine(serviceCreator)) {
-    tempActor.state = serviceCreator.initialState;
+    tempActor.state = serviceCreator.getInitialState(
```

Comment:
We can probably refactor this later, but:

```js
serviceCreator.withContext(mapContext(...)).initialState
```

Might be more accurate, no?

Reply:
good suggestion, I've applied the change